### DISCUSSION TITLE
Implement a way to stop traversal

### DISCUSSION THREAD

**Discussion by giuseppeg** [posthtml/posthtml#346] [Ideas]
In some cases it would be useful to stop a tree traversal when a node is found. 

Do you think it would make sense to pass an object with a `stop` method to `walk` and `match` which stops the recursion in `traverse` when called?

This is likely just a nice to have but in some circumstances it could help reduce overhead.

**Comment by Scrum** [comment]
Sounds good, can you describe a couple of examples of how you see it?
